modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/DotExpanderProcessorTests.java

```diff
@@ -143,4 +143,41 @@ public void testEscapeFields_path() throws Exception {
         assertThat(document.getFieldValue("field.foo.bar.baz", String.class), equalTo("value"));
     }
 
+
+    public void testEscapeFields_doNotingIfFieldNotInSourceDoc() throws Exception {
+        //asking to expand a (literal) field that is not present in the source document
+        Map<String, Object> source = new HashMap<>();
+        source.put("foo.bar", "baz1");
+        IngestDocument document = new IngestDocument(source, Collections.emptyMap());
+        //abc.def does not exist in source, so don't mutate document
+        DotExpanderProcessor processor = new DotExpanderProcessor("_tag", null, "abc.def");
+        processor.execute(document);
+        //hasField returns false since it requires the expanded form, which is not expanded since we did not ask for it to be
+        assertFalse(document.hasField("foo.bar"));
+        //nothing has changed
+        assertEquals(document.getSourceAndMetadata().get("foo.bar"), source.get("foo.bar"));
```

Comment:
is this necessary given the next line?

Reply:
I guess, are any of the other assertions necessary given that we are checking that source has not changed at all in this case (and no metadata was added).

Reply:
actually, comparing to source is mis-leading since source is the object that is getting changed, so `assertEquals(document.getSourceAndMetadata(), source);` is like asserting 1==1. 

I removed the assertions against source and kept the specific assertions. Thanks for spotting this.